teps/0135-coscheduling-pipelinerun-pods.md

```diff
@@ -0,0 +1,412 @@
+
+### Use Cases
+
+The following example is based on [pipeline#5275](https://github.com/tektoncd/pipeline/issues/5275):
+
+```yaml
+apiVersion: tekton.dev/v1
+kind: PipelineRun
+spec:
+  pipelineSpec:
+    workspaces:
+    - name: source
+    - name: cache
+    tasks:
+    - name: clone
+      taskRef:
+        name: git-clone
+      workspaces:
+      - name: source
+    - name: build
+      taskRef:
+        name: kaniko-build
+      workspaces:
+      - name: source
+      - name: cache
+  workspaces:
+  - name: source
+    volumeClaimTemplate:
+      spec:
+        accessModes:
+          - ReadWriteOnce
+        resources:
+          requests:
+            storage: 1Gi
+  - name: cache
+    persistentVolumeClaim:
+      claimName: build-cache-pvc
+```
+
+In this example, with the affinity assistant turned on, the TaskRun created for the Pipeline Task `kaniko-build` will fail validation. With the affinity assistant turned off, if the two volumes are associated with different nodes, the pod for the build TaskRun will be unschedulable.
+
+### Existing Workarounds
+
+Users can rewrite their PipelineRuns to use a single PVC-backed workspace and multiple sub-paths, with the affinity assistant enabled. See [this comment](https://github.com/tektoncd/pipeline/issues/3480#issuecomment-720442858) for more detail.
+
+### Requirements
+
+- Must be possible for the cluster operator to configure whether multiple PipelineRuns
+can run concurrently on the same node.
+- If it's not possible for all of a PipelineRun's pods to be scheduled to a node, the PipelineRun must still run to completion without operator intervention. See [Unschedulable Nodes](#unschedulable-nodes) for more detail.
+
+## Proposal
+
+PipelineRun pods can be scheduled to the same node using a mechanism similar to the existing affinity assistant: creating a placeholder pod and setting [inter-pod affinity](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#inter-pod-affinity-and-anti-affinity) for TaskRun pods to the placeholder pod. While the existing affinity assistant creates one placeholder pod per workspace, the proposed solution will create one placeholder pod per PipelineRun.
+
+## Design Details
+
+### Configuration
+
+Currently, the affinity assistant is enabled by default, and disabled by setting the boolean feature flag "disable-affinity-assistant" to true.
+
+This TEP proposes creating a new feature flag, "coscheduling". The options will be:
+- "disabled": The default value.
+- "coschedule-workspaces": The existing affinity assistant behavior. This option runs all of a PipelineRun's pods sharing the same workspace on the same node.
+- "coschedule-pipelineruns": This option runs all of a PipelineRun's pods on the same node.
+- "isolate-pipelineruns": This option runs all of a PipelineRun's pods on the same node, and only allows one PipelineRun to run on a node at a time.
```

Comment:
Any reason to tackle this one right now ? (aka I see less use-case than the one above)

Reply:
Fair